lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -30,18 +39,320 @@ const ignoredPathComponents = [
+          if (!value.includes(placeholder)) {
+            throw Error(`Example missing ICU replacement in message "${message}"`);
+          }
+          examples[placeholder.substring(1, placeholder.length - 1)] = message;
+        }
+      }
+      // Make sure all ICU vars have examples
+      while ((matches = findIcu.exec(value)) !== null) {
+        const varName = matches[1];
+        if (!examples[varName]) {
+          throw Error(`Variable '${varName}' is missing example comment in message "${value}"`);
+        }
+      }
+
+      // Make sure description is not empty
+      if (description.length === 0) throw Error(`Empty @description for message "${value}"`);
+      return {description, examples};
+    }
+
+    const description = comment.value.replace('*', '').trim();
+
+    // Make sure all ICU vars have examples
+    if (value.match(findIcu)) {
+      throw Error(`Variable '${value.match(/.*\{(\w+)\}.*/)[1]}' ` +
+        `is missing example comment in message "${value}"`);
+    }
+
+    // Make sure description is not empty
+    if (description.length === 0) throw Error(`Empty description for message "${value}"`);
+
+    // The entire comment is the description, so return everything.
+    return {description};
+  }
+  throw Error(`No Description for message "${value}"`);
+}
+
+/**
+ * Take a series of LHL format ICU messages and converts them
+ * to CTC format by replacing {ICU} and `markdown` with
+ * $placeholders$. Functional opposite of `bakePlaceholders`. This is commonly
+ * called as one of the first steps in translation, via collect-strings.js.
+ *
+ * Converts this:
+ * messages: {
+ *  "lighthouse-core/audits/seo/canonical.js | explanationDifferentDomain" {
+ *    "message": "Points to a different domain ({url})",
+ *    },
+ *  },
+ * }
+ *
+ * Into this:
+ * messages: {
+ *  "lighthouse-core/audits/seo/canonical.js | explanationDifferentDomain" {
+ *    "message": "Points to a different domain ($ICU_0$)",
+ *    "placeholders": {
+ *      "ICU_0": {
+ *        "content": "{url}",
+ *        "example": "https://example.com/"
+ *      },
+ *    },
+ *  },
+ * }
+ *
+ * Throws if the message violates some basic sanity checking.
+ *
+ * @param {string} message
+ * @param {Record<string, string>} examples
+ * @returns {ICUMessageDefn}
+ */
+function convertMessageToPlaceholders(message, examples = {}) {
+  // Basically the same as markdown parsing in dom.js
+  const icu = {
+    message,
+    placeholders: {},
+  };
+
+  // Process each placeholder type
+  _processPlaceholderMarkdownCode(icu);
+
+  _processPlaceholderMarkdownLink(icu);
+
+  _processPlaceholderComplexIcu(icu);
+
+  _processPlaceholderDirectIcu(icu, examples);
+
+  return icu;
+}
+
+/**
+ * Convert markdown code blocks into placeholders with examples.
+ *
+ * @param {ICUMessageDefn} icu
+ */
+function _processPlaceholderMarkdownCode(icu) {
+  // Check that number of backticks is even.
+  if ((icu.message.split('`').length - 1) % 2 !== 0) {
```

Comment:
seems a bit indirect. how about

```suggestion
  const match = icu.message.match(/\`/g);
  if (match && match.length % 2 !== 0) {
```